tutor/src/components/change-student-id.cjsx

```diff
@@ -0,0 +1,100 @@
+_ = require 'lodash'
+React = require 'react'
+BS = require 'react-bootstrap'
+
+LoadableItem = require './loadable-item'
+{ ChangeStudentIdForm } = require 'shared'
+{ TransitionActions, TransitionStore } = require '../flux/transition'
+{ CourseStore, CourseActions } = require '../flux/course'
+{ StudentIdStore, StudentIdActions, ERROR_MAP } = require '../flux/student-id'
+Icon = require './icon'
+
+module.exports = React.createClass
+  contextTypes:
+    router: React.PropTypes.func
+
+  onCancel: -> @goBack()
+
+  componentWillMount: ->
+    {courseId} = @context.router.getCurrentParams()
+    CourseStore.once('course.loaded', @update)
+    CourseActions.load(courseId)
+
+  componentWillUnmount: ->
+    CourseStore.off('course.loaded', @update)
+    StudentIdStore.off('student-id-saved', @saved)
+    StudentIdStore.off('student-id-error', @update)
+
+  update: ->
+    @setState({})
+
+  goBack: ->
+    historyInfo = TransitionStore.getPrevious(@context.router)
+    url = historyInfo.path or 'dashboard'
+    @context.router.transitionTo(url)
+
+  onSubmit: (newStudentId) ->
+    {courseId} = @context.router.getCurrentParams()
+
```

Comment:
any chance either here, in the two `onSubmit`s being fed from `tutor` and `coach` to check for where it's the same id before submitting? or in the `onSubmit` in the `shared` component, although that would make it tricky for any other side-effect from a "successful" submit to be called